#! /usr/bin/env python
# Task Long configuration

PACKAGE='robot_control_base'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#       Name                    Type            Description
#       Default    Min   Max
gen.add("velocity",      double_t, 0,    "Constant velocity",  0.5, 0.0, 10.0)
gen.add("k_y",           double_t, 0,    "Gain for lateral control",  1.0, -20.0, 20.0)
gen.add("k_theta",       double_t, 0,    "Gain for angular control",  1.0, -20.0, 20.0)
gen.add("max_angular_velocity",  double_t, 0,    "Max allowed angular velocity",  0.5, 0.0, 5.0)
gen.add("max_y_error",   double_t, 0,    "Saturation value for lateral error",  2.0, 0.0, 10.0)
gen.add("dist_threshold",double_t, 0,    "Distance at which a the target is considered reached",  0.1, 0.0, 10.0)


exit(gen.generate(PACKAGE, "robot_control_base", "FollowLine"))

